static/app/components/replays/replayContext.tsx

```diff
@@ -551,20 +553,31 @@ function ProviderNonMemo({
 
   useEffect(() => {
     const handleVisibilityChange = () => {
-      if (document.visibilityState !== 'visible') {
+      if (document.visibilityState !== 'visible' && replayerRef.current) {
         togglePlayPause(false);
       }
     };
 
+    document.addEventListener('visibilitychange', handleVisibilityChange);
+
+    return () => {
+      document.removeEventListener('visibilitychange', handleVisibilityChange);
+    };
+  }, [togglePlayPause]);
```

Comment:
I split this into two `useEffects`, I don't think this listener needs to be added based on `events` and `videoEvents`